iroh/src/endpoint/quic.rs

```diff
@@ -484,3 +543,177 @@ impl QuicTransportConfig {
+    pub fn max_incoming(&mut self, max_incoming: usize) -> &mut Self {
+        self.inner.max_incoming(max_incoming);
+        self
+    }
+
+    /// Maximum number of received bytes to buffer for each [`Incoming`]
+    ///
+    /// An [`Incoming`] comes into existence when an incoming connection attempt
+    /// is received and stops existing when the application either accepts it or otherwise disposes
+    /// of it. This limit governs only packets received within that period, and does not include
+    /// the first packet. Packets received in excess of this limit are dropped, which may cause
+    /// 0-RTT or handshake data to have to be retransmitted.
+    ///
+    /// The default value is set to 10 MiB--an amount such that in most situations a client would
+    /// not transmit that much 0-RTT data faster than the server handles the corresponding
+    /// [`Incoming`].
+    ///
+    /// [`Incoming`]: crate::endpoint::Incoming
+    pub fn incoming_buffer_size(&mut self, incoming_buffer_size: u64) -> &mut Self {
+        self.inner.incoming_buffer_size(incoming_buffer_size);
+        self
+    }
+
+    /// Maximum number of received bytes to buffer for all [`Incoming`]
+    /// collectively
+    ///
+    /// An [`Incoming`] comes into existence when an incoming connection attempt
+    /// is received and stops existing when the application either accepts it or otherwise disposes
+    /// of it. This limit governs only packets received within that period, and does not include
+    /// the first packet. Packets received in excess of this limit are dropped, which may cause
+    /// 0-RTT or handshake data to have to be retransmitted.
+    ///
+    /// The default value is set to 100 MiB--a generous amount that still prevents memory
+    /// exhaustion in most contexts.
+    ///
+    /// [`Incoming`]: crate::endpoint::Incoming
+    pub fn incoming_buffer_size_total(&mut self, incoming_buffer_size_total: u64) -> &mut Self {
+        self.inner
+            .incoming_buffer_size_total(incoming_buffer_size_total);
+        self
+    }
+
+    /// Object to get current [`SystemTime`]
+    ///
+    /// This exists to allow system time to be mocked in tests, or wherever else desired.
+    ///
+    /// Defaults to [`quinn::StdSystemTime`], which simply calls [`SystemTime::now()`](std::time::SystemTime::now).
+    ///
+    /// [`SystemTime`]: std::time::SystemTime
+    pub fn time_source(&mut self, time_source: Arc<dyn TimeSource>) -> &mut Self {
+        self.inner.time_source(time_source);
+        self
+    }
+
+    /// Create a server config with the given [`CryptoServerConfig`]
+    ///
+    /// Uses a randomized handshake token key.
+    pub fn with_crypto(crypto: Arc<dyn CryptoServerConfig>) -> Self {
+        let mut inner = quinn::ServerConfig::with_crypto(crypto);
+        let transport = QuicTransportConfig::default();
+        inner.transport_config(transport.to_arc());
+        Self { inner, transport }
+    }
```

Comment:
Lets remove this one for now, you can create something that isn't iroh anymore. Though you *can* also do interesting stuff, but let's handle those as feature requests.